### ISSUE TITLE
saltutil.sync_xxx does not work correctly

### ISSUE THREAD

**Issue by czhong111** [saltstack/salt#47582]
### Description of Issue/Question
When I upgrade salt-master from version 2015 to 2017.7.5-1 version, and find that saltutil.sync_xxx commands does not work as expected, and module/grains in /srv/salt/(_grains|_modules) files have not been transfered to  /var/cache/salt/minion/extmods on minoins.

  
### Setup

master setupfile
cat /etc/salt/master

```
id: saltcmdevslb05
master_id: saltcmdevslb05
interface: saltcmdevslb05
worker_threads: 4
file_roots:
  base:
   - /srv/salt
file_ignore_regex:
  - '/\.svn($|/)'
pillar_roots:
  base:
   - /srv/pillar
cachedir: /opt/cache/salt/master
#log_level: trace
log_file: /opt/log/salt/master
log_level_logfile: debug
timeout: 30
token_expire: 90000
keep_jobs: 1
zmq_filtering: true
```

on minion side
#cat /etc/salt/minion

```
master: saltcmdevslb05
id: web1
zmq_filtering: True
```

### Steps to Reproduce Issue
Run salt command as followed

```
salt web1 saltutil.sync_grains -l debug
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/master.d/api.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/api.conf
[DEBUG   ] Including configuration from '/etc/salt/master.d/eauth.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/eauth.conf
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Including configuration from '/etc/salt/master.d/api.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/api.conf
[DEBUG   ] Including configuration from '/etc/salt/master.d/eauth.conf'
[DEBUG   ] Reading configuration from /etc/salt/master.d/eauth.conf
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/master', 'saltcmdevslb05', 'tcp://saltcmdevslb05:4506', 'clear')
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] LazyLoaded local_cache.get_load
[DEBUG   ] Reading minion list from /opt/cache/salt/master/jobs/e0/f30870cfce6a4801ebaf5cfcb4540644bb853116727d097b6d68534d282086/.minions.p
[DEBUG   ] get_iter_returns for jid 20180510104344226193 sent to set(['web1']) will timeout at 10:44:14.235862
[DEBUG   ] jid 20180510104344226193 return from web1
[DEBUG   ] return event: {'web1': {'jid': '20180510104344226193', 'retcode': 0, 'ret': []}}
[DEBUG   ] LazyLoaded nested.output
web1:
[DEBUG   ] jid 20180510104344226193 found all minions set(['web1'])
```

### Versions Report

master version
OS: Centos 7.3
Salt Version:
           Salt: 2017.7.5
 
Dependency Versions:
           cffi: 1.6.0
       cherrypy: unknown
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-514.26.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core

minion version
**Some minion vms are**
os: centos 7.3
salt version: 2015.5.11-1 /2017.7.5-1 
**others are**
os:rhel6.3
salt version: salt-2015.5.3-5

**Comment by Ch3LL** [comment]
if you try this against a 2017.7.5 minion does it work?

**Comment by Ch3LL** [comment]
oh nevermind i just re-read your salt versions and noticed you also listed that some of hte minions are 2017.7.5 let me re-read more carefully

**Comment by Ch3LL** [comment]
I am not able to replicate this even when upgrading from 2015.5.11 to 2017.7.5. Can you paste the sanitized output from `salt-call saltutil.sync_grains -ldebug; ls /var/cache/salt/minion/extmods/grains` from the minion?

**Comment by dhs-rec** [comment]
I'm also having problems syncing new _states/ to Minions as long as they exist in a development branch of a gitfs repo only. saltutil.sync_all (or sync_states) will not sync the new state unless it's been made visible in the base environment, too, even if the saltenv is explicitely given on the command line.

**Comment by Ch3LL** [comment]
@czhong111 looking at your log output it seems it should be created the directory from the log here:

```
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'web1', 'tcp://saltcmdevslb05:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'web1', 'tcp://saltcmdevslb05:4506')
[TRACE   ] Inserted key into loop_instance_map id 40967272 for key ('/etc/salt/pki/minion', 'web1', 'tcp://saltcmdevslb05:4506', 'aes
') and process 26985[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base)
[INFO    ] Caching directory '_grains/' for environment 'base'
[DEBUG   ] Local cache dir: '/var/cache/salt/minion/files/base/_grains'
```

The creation of the directory is from the code here: https://github.com/saltstack/salt/blob/v2017.7.5/salt/utils/extmods.py#L71  It should be logging an error if it can't create the directory which i'm not seeing in your output. 


@czhong111 are you possibly setting a different cache dir for the minion?

Also can you try wiping away the cache directory for the minion and trying again. 

@dhs-rec it seems your issue is different from this as we are in the base environment here. can you open an issue with more details?

**Comment by dhs-rec** [comment]
@Ch3LL, done: #47763

**Comment by czhong111** [comment]
@Ch3LL, in the master log, I found the err log.
`2018-05-23 10:15:36,163 [salt.fileserver  :266 ][DEBUG   ][5258] File matching file_ignore_regex. Skipping: zabbix_physical/.svn/tmp/prop-base
2018-05-23 10:15:36,187 [salt.master      :1644][ERROR   ][5258] Error in function _file_list:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/master.py", line 1634, in run_func
    ret = getattr(self, func)(load)
  File "/usr/lib/python2.7/site-packages/salt/fileserver/__init__.py", line 756, in file_list
    ret.update(self.servers[fstr](load))
  File "/usr/lib/python2.7/site-packages/salt/fileserver/roots.py", line 431, in file_list
    return _file_lists(load, 'files')
  File "/usr/lib/python2.7/site-packages/salt/fileserver/roots.py", line 407, in _file_lists
    _add_to(ret['files'], path, root, files)
  File "/usr/lib/python2.7/site-packages/salt/fileserver/roots.py", line 343, in _add_to
    log.trace('roots: Processing %s', abs_path)
  File "/usr/lib/python2.7/site-packages/salt/log/mixins.py", line 35, in trace
    self.log(getattr(logging, 'TRACE', 5), msg, *args, **kwargs)
  File "/usr/lib64/python2.7/logging/__init__.py", line 1213, in log
    raise TypeError("level must be an integer")
  File "/usr/lib/python2.7/site-packages/salt/log/setup.py", line 327, in _log
    self, level, msg, args, exc_info=exc_info, extra=extra
  File "/usr/lib64/python2.7/logging/__init__.py", line 1267, in _log
    fn, lno, func = "(unknown file)", 0, "(unknown function)"
  File "/usr/lib/python2.7/site-packages/salt/log/setup.py", line 361, in makeRecord
    exc_info, func)
  File "/usr/lib/python2.7/site-packages/salt/log/setup.py", line 208, in __init__
    self.colormsg = '%s%s%s' % (cmsg, self.getMessage(), reset)
  File "/usr/lib64/python2.7/logging/__init__.py", line 328, in getMessage
    msg = str(self.msg)
**UnicodeDecodeError: 'ascii' codec can't decode byte 0xe5 in position 38: ordinal not in range(128**)
`


and in master setting file, the file_ignore_regex may cause the error, '/\.svn($|/)‘ cannot run correctly in python 2.7 but work in python2.6
`file_ignore_regex:
  - '/\.svn($|/)'
`

**Comment by Ch3LL** [comment]
I am able to replicate this and have verified that this issue will be resolved in the upcoming 2017.7.6 release  and its currently fixed in the 2018.3.0 release.

**Comment by czhong111** [comment]
thanks a lot
